### PR TITLE
[wip] Fix interface with the new package check

### PR DESCRIPTION
**Pull request by alexAubin**
So that's a super-drafty attempt to adapt CI_package_check for https://github.com/YunoHost/package_check/pull/92 ... Unfortunately there's flamby everywhere and I'm not sure I want to go into deploying this beast anywhere ...

### CODE REVIEW

**Comment by alexAubin** [comment]
So that should be in a somewhat testable state ...

I guess I should have a look into how to actually deploy this ...

**Comment by alexAubin** [comment]
So of course I ended up refactoring the entire thing because what the hell >_> ...

I'm starting to be kinda ~done with this and have tested this both for the "regular" CI and dev CI

Trying to make a summary of what changed : 
- the install procedure now consist in running the obvious "install.sh" which will deploy everything (yunohost + yunorunner + chroot dir if needed + lxd)
- Some optimization during install.sh : I disabled all unecessary services to save up RAM because I was running my test on a Scaleway DEV-1S with 2GB ram and for example testing weblate resulted in super low RAM ...
- no more madness about analyseCI.sh vs pcheckCI.sh with epic lock shitstorms ... now yunorunner call analyseCI.sh which runs the entire job **as a non-root user!** because there's actually no necessity to be root (just need to be in the `lxd` group). analyseCI.sh is now pretty readable and only 200 lines of bash.
- no more madness about sending jobs to "other CI_package_check instances" e.g. for ARM tests ... c.f. the tests from Kayou, we should be able to run ARM tests using cross-arch LXCs
- no more madness about parsing the arch / branch from the test name with weird syntax like `.-(~*ARM*~)-.` ... now the CI_package_check config file contains an `ARCH=amd64` and `BRANCH=stable` that's applied to all jobs no matter what.
- removed a bunch of scripts which were unecessarily complex such as comparing the level + publishing the results into a proper json after the tests end .. now it's just a handful of bash lines at the end of analyseCI.sh
- the output of the test is now pretty much "real time" ... no more refresh every ~10s only
- the dev CI is now handled by yunorunner as well - just gotta fix minor bug in the home page. Ideally oneday™ we could have a small form to add job using a token + git url instead of having to register an SSH key + run a weird bash script


Remaining quirks
- Didn't test XMPP notifications ... supposedly just gotta provide the password
- After testing all the fancy LXD storage backend/drivers (btrfs, lvm, and couldnt installed zfs) i settled with the boring "dir" driver which at least doesn't have much surprises like "lol i don't support swapfiles" (btrfs) or "i randomly become readonly" (lvm) ... Still I think I spotted some issue where swap files, if not properly deleted by the app, remain mounted on the host and make LXD unwilling to delete the container, resulting in more explosion. But should be fixable in package_check by identifying remaining mounted swap files from the container I suppose ...
- LXD can start exploding when very few ressources are available on the system (c.f. testing weblate which for some reasons eats a shitload of ressource ...)
- Properly cleaning the lock when a job is canceled ? Supposedly fixed by https://github.com/YunoHost/yunorunner/pull/9
- Did not test/reimplement yet the "automatically trigger a job for high quality apps PRs" but sounds easily doable..
- Didn't check if badges work as expected

File: install.sh [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+    # ci_user will be the one launching job, gives it permission to run lxd commands
+    usermod -a -G lxd $ci_user
+
+    # We need a home for the "su" command later ?
+    mkdir -p /home/$ci_user
+    chown -R $ci_user /home/$ci_user
+
+    # Stupid hack because somehow ipv6 is interfering
+    echo "$(dig +short A devbaseimgs.yunohost.org | tail -n 1) devbaseimgs.yunohost.org" >> /etc/hosts
```
Comment (kay0u) [inline install.sh:144]:
```suggestion
```

Fixed now

File: analyseCI.sh [hunk 2]
```diff
@@ -1,29 +1,22 @@
...
-#=================================================
-
 if [ "${0:0:1}" == "/" ]; then script_dir="$(dirname "$0")"; else script_dir="$(echo $PWD/$(dirname "$0" | cut -d '.' -f2) | sed 's@/$@@')"; fi
 
-#=================================================
-# Check the number of arguments
-#=================================================
+cd $script_dir
 
 if [ $# -ne 2 ]
```
Comment (kay0u) [inline analyseCI.sh:7]:
```suggestion
if [ $# -ne 3 ]
```

cf. https://github.com/YunoHost/yunorunner/commit/470fb02022e3b923519cd5786eac0ce16ec41f16

File: install.sh [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    echo_bold "> Installing dependencies..."
+    apt-get update
+    apt-get install -y curl wget git python3-pip lynx jq snapd
+    pip3 install xmpppy
+    
+    # Install Package check if it isn't an ARM only CI.
+    git clone https://github.com/YunoHost/package_check "./package_check" -b cleanup-3 --single-branch
+
+    # Download the app badges
+    ./lib/badges/get_badges.sh
```
Comment (kay0u) [inline install.sh:57]:
```suggestion
    ./lib/get_badges.sh
```

File: install.sh [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+    echo_bold "> Installing dependencies..."
+    apt-get update
+    apt-get install -y curl wget git python3-pip lynx jq snapd
+    pip3 install xmpppy
+    
+    # Install Package check if it isn't an ARM only CI.
+    git clone https://github.com/YunoHost/package_check "./package_check" -b cleanup-3 --single-branch
+
+    # Download the app badges
+    ./lib/get_badges.sh
```
Comment (kay0u) [inline install.sh:57]:
```suggestion
    ./badges/get_badges.sh
```

File: analyseCI.sh [hunk 5]
```diff
@@ -4,7 +4,7 @@ if [ "${0:0:1}" == "/" ]; then script_dir="$(dirname "$0")"; else script_dir="$(
 
 cd $script_dir
 
-if [ $# -ne 2 ]
```
Comment (yalh76) [inline analyseCI.sh:17]: Why changing to 3 ? what is the third argument ? In the [readme ](https://github.com/YunoHost/CI_package_check/blob/master/README.md) there is still 2 arguments to `/PATH/analyseCI.sh`.

When I launch something like:
`/home/CI_package_check_buster/analyseCI.sh "/home/package_check/peertube_ynh" "peertube_yalh76_buster_testing"` I have as result : `This script need to take in argument the package which be tested and the name of the test.`

Comment (alexAubin) [reply analyseCI.sh:17]: Erh yeah indeed .. the third arg is supposed to be the test id provided by yunorunner .. it's in fact only used to later create a nice link in the XMPP notification (e.g. ci-apps-foobar.yunohost.org/job/$job_id). As a temporary workaround, you can feed it empty string and that should be okay

**Pull request merged**